Possible error in 1.2.0 around macro extension

When [dlme-transform](https://github.com/sul-dlss/dlme-transform) was updated to Traject 3.2.0 and TrajectPlus 1.2.0, it stopped working:

```shell
$ docker run --rm --lots --of --arguments suldlss/dlme-transform:latest stanford/maps/data/kj751hs0595.mods

[ERROR] Error loading configuration file config/mods_config.rb:11 NoMethodError:undefined method `extended' for #<Traject::Indexer:0x0000564b796c1bd0>
Did you mean?  extend                                                                                                                                 
/usr/local/bundle/gems/traject-3.2.0/lib/traject/indexer.rb:231:in `rescue in block in load_config_file': Error loading configuration file config/mods
_config.rb:11 NoMethodError:undefined method `extended' for #<Traject::Indexer:0x0000564b796c1bd0> (Traject::Indexer::ConfigLoadError)                
Did you mean?  extend                                                                                                                                 
        from /usr/local/bundle/gems/traject-3.2.0/lib/traject/indexer.rb:228:in `block in load_config_file'                                           
        from /usr/local/bundle/gems/traject-3.2.0/lib/traject/indexer.rb:227:in `open'                                                                
        from /usr/local/bundle/gems/traject-3.2.0/lib/traject/indexer.rb:227:in `load_config_file'                                                    
        from /opt/traject/lib/transformer.rb:33:in `block (2 levels) in transformer'                                                                  
        from /opt/traject/lib/transformer.rb:33:in `each'                                                                                             
        from /opt/traject/lib/transformer.rb:33:in `block in transformer'                                                                             
        from /opt/traject/lib/transformer.rb:32:in `tap'                                                                                              
        from /opt/traject/lib/transformer.rb:32:in `transformer'                                                                                      
        from /opt/traject/lib/transformer.rb:25:in `transform'                                                                                        
        from /opt/traject/lib/cli.rb:75:in `block in transform_all'                                                                                   
        from /opt/traject/lib/cli.rb:69:in `each'                                                                                                     
        from /opt/traject/lib/cli.rb:69:in `transform_all'                                                                                            
        from /opt/traject/lib/cli.rb:53:in `transform'                                                                                                
        from /usr/local/bundle/gems/thor-0.20.3/lib/thor/command.rb:27:in `run'                                                                       
        from /usr/local/bundle/gems/thor-0.20.3/lib/thor/invocation.rb:126:in `invoke_command'                                                        
        from /usr/local/bundle/gems/thor-0.20.3/lib/thor.rb:387:in `dispatch'                                                                         
        from /usr/local/bundle/gems/thor-0.20.3/lib/thor/base.rb:466:in `start'                                                                       
        from exe/transform:8:in `<main>'                                                            
```

When I pin TrajectPlus to v1.1.1, the above error goes away.

Here is where our indexer is set up: https://github.com/sul-dlss/dlme-transform/blob/master/lib/transformer.rb#L32-L40

This line is the one on which the exception is raised: https://github.com/sul-dlss/dlme-traject/blob/master/mods_config.rb#L11

This is the code that has the `NoMethodError`: https://github.com/sul-dlss/traject_plus/blob/master/lib/traject_plus/macros/mods.rb#L13 presumably because the `mod` argument passed to `self.extended` is an instance of a `Traject::Indexer` class and class instances are not modules and thus don't respond to `self.extended`. Is this a bug in TrajectPlus or a misconfiguration in one of the DLME codebases?

--- a/lib/traject_plus/macros/fgdc.rb
+++ b/lib/traject_plus/macros/fgdc.rb
@@ -6,9 +6,7 @@ module TrajectPlus
     module FGDC
       NS = { fgdc: 'http://www.fgdc.gov/metadata/fgdc-std-001-1998.dtd' }.freeze
 
-      def self.extended(mod)
-        mod.extended Traject::Macros::NokogiriMacros
-      end
+      include Traject::Macros::NokogiriMacros
 
       # @param xpath [String] the xpath query expression
       def extract_fgdc(xpath)

--- a/lib/traject_plus/macros/mods.rb
+++ b/lib/traject_plus/macros/mods.rb
@@ -9,13 +9,11 @@ module TrajectPlus
              dc: 'http://purl.org/dc/elements/1.1/',
              xlink: 'http://www.w3.org/1999/xlink' }.freeze
 
-      def self.extended(mod)
-        mod.extended Traject::Macros::NokogiriMacros
-      end
+      include Traject::Macros::NokogiriMacros
 
       # @param xpath [String] the xpath query expression
       def extract_mods(xpath)
-        extract_xpath(xpath, NS)
+        extract_xpath(xpath, ns: NS)
       end
     end
   end

--- a/lib/traject_plus/macros/tei.rb
+++ b/lib/traject_plus/macros/tei.rb
@@ -5,9 +5,7 @@ module TrajectPlus
     module Tei
       NS = { tei: 'http://www.tei-c.org/ns/1.0' }.freeze
 
-      def self.extended(mod)
-        mod.extended Traject::Macros::NokogiriMacros
-      end
+      include Traject::Macros::NokogiriMacros
 
       # @param xpath [String] the xpath query expression
       def extract_tei(xpath)

--- a/spec/lib/traject_plus/macros_spec.rb
+++ b/spec/lib/traject_plus/macros_spec.rb
@@ -11,15 +11,25 @@ RSpec.describe TrajectPlus::Macros do
   end
 
   describe '#transform_values' do
-    let(:doc) { Nokogiri::XML('<xml><foo> bar </foo></xml>') }
+    let(:doc) do
+      Nokogiri::XML(
+        <<-XML
+          <?xml version="1.0" encoding="UTF-8"?>
+          <mods xmlns:xlink="http://www.w3.org/1999/xlink" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns="http://www.loc.gov/mods/v3" xsi:schemaLocation="http://www.loc.gov/mods/v3 http://www.loc.gov/standards/mods/v3/mods-3-3.xsd" version="3.3">
+            <genre>bar</genre>
+          </mods>
+        XML
+      )
+    end
+
     context 'with lambdas with 2 and 3 arity' do
       it 'runs both' do
-        indexer.extend Traject::Macros::NokogiriMacros
+        indexer.extend TrajectPlus::Macros::Mods
         indexer.instance_eval do
           to_field 'some_field' do |_record, accumulator, context|
             accumulator << transform_values(
               context,
-              'wr_id' => [extract_xpath('//foo'), strip]
+              'wr_id' => [extract_xpath('//genre'), strip]
             )
           end
         end

--- a/spec/lib/traject_plus/macros_spec.rb
+++ b/spec/lib/traject_plus/macros_spec.rb
@@ -1,4 +1,5 @@
 # frozen_string_literal: true
+
 require 'spec_helper'
 
 RSpec.describe TrajectPlus::Macros do
@@ -29,12 +30,14 @@ RSpec.describe TrajectPlus::Macros do
           to_field 'some_field' do |_record, accumulator, context|
             accumulator << transform_values(
               context,
-              'wr_id' => [extract_xpath('//genre'), strip]
+              'wr_id' => [extract_xpath('//mods:genre', ns: TrajectPlus::Macros::Mods::NS), strip]
             )
           end
+          to_field 'other_field', extract_mods('/*/mods:genre')
         end
 
-        expect(indexer.map_record(doc)).to eq('some_field' => [{ 'wr_id' => ['bar'] }])
+        expect(indexer.map_record(doc)).to eq('some_field' => [{ 'wr_id' => ['bar'] }],
+                                              'other_field' => ['bar'])
       end
     end
   end

